GitUI/CommandsDialogs/FormPull.cs

```diff
@@ -216,8 +222,35 @@ private void BindRemotesDropDown(string selectedRemoteName)
             }
         }
 
-        public DialogResult PullAndShowDialogWhenFailed(IWin32Window owner)
+        public DialogResult PullAndShowDialogWhenFailed(IWin32Window owner, string remote, AppSettings.PullAction pullAction)
         {
+            // Special case for FetchPrune and FetchPruneAll to make sure user confirms the action.
+            string messageBoxTitle = null;
+            if (pullAction == AppSettings.PullAction.FetchPrune)
+            {
+                messageBoxTitle = string.Format(_pruneFromCaption.Text, remote);
+            }
+
+            if (pullAction == AppSettings.PullAction.FetchPruneAll)
+            {
+                messageBoxTitle = string.Format(_pruneFromCaption.Text, AllRemotes);
```

Comment:
Can we express the difference under `PullAction.FetchPruneAll` but supply `AllRemotes` if `remote == null`?

Reply:
Sorry, I'm not sure if I understand you correctly. Do you want to simply improve the readability of this method?

Or do you want to eliminate `PullAction.FetchPrune` by using `PullAction.FetchPruneAll` for both operations "Fetch and prune" and "Fetch and prune all"? Something like:
```
    if (pullAction == AppSettings.PullAction.FetchPruneAll)
    {
        if (remote.IsNullOrEmpty())
        {
            // Fetch and prune AllRemotes
        }
        else
        {
            // Fetch and prune remote
        }

```

Reply:
Apologies for being unclear.
I'd like to eliminate `FetchPrune` altogether, the implementation looks similar to `FetchPruneAll`.

Reply:
Indeed, they are very similar. I can make it work without `FetchPrune`.

Reply:
Done, please have a look.